src/libs/ck-libs/allGather/allGather.hh

```diff
@@ -0,0 +1,65 @@
+#pragma once
+
+#include "allGather.decl.h"
+#include <cassert>
+#include <cstdlib>
+#include <cstring>
+#include <map>
+#include <random>
+#include <utility>
+#include <vector>
+
+class allGatherMsg : public CMessage_allGatherMsg
+{
+};
+
+enum allGatherType
+{
+  ALL_GATHER_RING,
+  ALL_GATHER_HYPERCUBE,
+  ALL_GATHER_FLOODING
+};
+
+class AllGather : public CBase_AllGather
+{
+private:
+  int k{};
+  int n{};
+  int idx{};
+  long int* store;
+  int numRecvMsg{};
+  CkCallback lib_done_callback;
+  allGatherType type;
+  int numHypercubeIter{};
+  bool HypercubeRecursiveDoubling{};
+  int iter;
+  int HypercubeToSend;
+  std::vector<std::vector<int>> graph{};
+  std::map<int, bool> recvFloodMsg{};
+  int randCounter{};
+  std::vector<int> hyperCubeIndx{};
+  std::vector<CkNcpyBuffer> hyperCubeStore{};
+  allGatherMsg* msg = new allGatherMsg;
```

Comment:
I think we generally prefer dynamic memory allocation be done in the constructor rather than here